addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRule.java

```diff
@@ -570,40 +627,165 @@ private boolean performCloseTagAttack(HtmlContext context, HttpMessage msg, Stri
+
+        if (alertStatus) {
+            newAlert()
+                    .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                    .setParam(param)
+                    .setAttack(attack)
+                    .setEvidence(attack)
+                    .setMessage(context.getMsg())
+                    .raise();
+
+            return true;
+        }
+
+        // simple direct attack in no quoted string
+        attack = "alert(1);//";
+        alertStatus = performAttackForScript(msg, param, attack);
+
+        if (alertStatus) {
+            newAlert()
+                    .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                    .setParam(param)
+                    .setAttack(attack)
+                    .setEvidence(attack)
+                    .setMessage(context.getMsg())
+                    .raise();
+
+            return true;
+        }
+
+        // simple direct attack in slash quoted string
+        attack = "x/;alert(1);//";
+        alertStatus = performAttackForScript(msg, param, attack);
+
+        if (alertStatus) {
+            newAlert()
+                    .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                    .setParam(param)
+                    .setAttack(attack)
+                    .setEvidence(attack)
+                    .setMessage(context.getMsg())
+                    .raise();
+
+            return true;
+        }
+
+        // simple direct attack in single quotes
+        attack = "';alert(1);//";
+        alertStatus = performAttackForScript(msg, param, attack);
+
+        if (alertStatus) {
+            newAlert()
+                    .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                    .setParam(param)
+                    .setAttack(attack)
+                    .setEvidence(attack)
+                    .setMessage(context.getMsg())
+                    .raise();
+
+            return true;
+        }
+
+        // escape backslash direct attack in doublequotes
+        attack = "\\\";alert(1);//";
+        alertStatus = performAttackForScript(msg, param, attack);
+
+        if (alertStatus) {
+            newAlert()
+                    .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                    .setParam(param)
+                    .setAttack(attack)
+                    .setEvidence(attack)
+                    .setMessage(context.getMsg())
+                    .raise();
+
+            return true;
+        }
+
+        // escape backslash direct attack in single quotes
+        attack = "\';alert(1);//";
+        alertStatus = performAttackForScript(msg, param, attack);
+
+        if (alertStatus) {
+            newAlert()
+                    .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                    .setParam(param)
+                    .setAttack(attack)
+                    .setEvidence(attack)
+                    .setMessage(context.getMsg())
+                    .raise();
+
+            return true;
+        }
```

Comment:
It seems like all the attacks in 1 should be able to fall through to a single raise alert/return true instead of repeating it over and over.

Reply:
I have updated the method accordingly.